storage/aptosdb/src/lib.rs

```diff
@@ -1006,100 +867,278 @@ impl AptosDB {
+            &ledger_metadata_batch,
+            &sharded_state_kv_batches,
+        )?;
+
+        let last_version = first_version + txns_to_commit.len() as u64 - 1;
+        ledger_metadata_batch
+            .put::<DbMetadataSchema>(
+                &DbMetadataKey::LedgerCommitProgress,
+                &DbMetadataValue::Version(last_version),
+            )
+            .unwrap();
+
+        let _timer = OTHER_TIMERS_SECONDS
+            .with_label_values(&["commit_state_kv_and_ledger_metadata___commit"])
+            .start_timer();
+        thread::scope(|s| {
+            s.spawn(|| {
                 self.ledger_db
-                    .transaction_db()
-                    .write_schemas(ledger_schema_batches.transaction_batch)
+                    .metadata_db()
+                    .write_schemas(ledger_metadata_batch)
                     .unwrap();
             });
-            s.spawn(|_| {
-                let _timer = OTHER_TIMERS_SECONDS
-                    .with_label_values(&["save_transactions_commit___transaction_info_commit"])
-                    .start_timer();
-                self.ledger_db
-                    .transaction_info_db()
-                    .write_schemas(ledger_schema_batches.transaction_info_batch)
+            s.spawn(|| {
+                self.state_kv_db
+                    .commit(last_version, sharded_state_kv_batches)
                     .unwrap();
             });
-            s.spawn(|_| {
+        });
+
+        Ok(())
+    }
+
+    fn commit_events(
+        &self,
+        txns_to_commit: &[impl Borrow<TransactionToCommit> + Sync],
+        first_version: Version,
+    ) -> Result<()> {
+        let _timer = OTHER_TIMERS_SECONDS
+            .with_label_values(&["commit_events"])
+            .start_timer();
+        let batch = SchemaBatch::new();
+        txns_to_commit
+            .par_iter()
+            .with_min_len(128)
+            .enumerate()
+            .try_for_each(|(i, txn_to_commit)| -> Result<()> {
+                self.event_store.put_events(
+                    first_version + i as u64,
+                    txn_to_commit.borrow().events(),
+                    &batch,
+                )?;
+
+                Ok(())
+            })?;
+        let _timer = OTHER_TIMERS_SECONDS
+            .with_label_values(&["commit_events___commit"])
+            .start_timer();
+        self.ledger_db.event_db().write_schemas(batch)
+    }
+
+    fn commit_transactions(
+        &self,
+        txns_to_commit: &[impl Borrow<TransactionToCommit> + Sync],
+        first_version: Version,
+    ) -> Result<()> {
+        let _timer = OTHER_TIMERS_SECONDS
+            .with_label_values(&["commit_transactions"])
+            .start_timer();
+        let chunk_size = 512;
+        txns_to_commit
+            .par_chunks(chunk_size)
+            .enumerate()
+            .try_for_each(|(chunk_index, txns_in_chunk)| -> Result<()> {
+                let batch = SchemaBatch::new();
```

Comment:
Does it really matter to use multiple batches? (and why does it matter only for transactions?)

Reply:
It's ~10% faster in my benchmark.. (I only did it for txn because it is relatively larger comparing to other data types)

btw, I don't like it.